### PR TITLE
WT-9781 Fix search near exact edge case with bounded cursor

### PR DESCRIPTION
**Pull request by clarissecheah**
- Fix edge case bug where search near was returning the wrong exact with bounded cursors.
- Added python test.

### CODE REVIEW

File: src/btree/bt_cursor.c [hunk 1]
```diff
@@ -1019,6 +1019,13 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
             __cursor_state_restore(cursor, &state);
         else {
             __wt_value_return(cbt, cbt->upd_value);
+            if (WT_CURSOR_BOUNDS_SET(cursor)) {
+                if (btree->type == BTREE_ROW)
+                    WT_ERR(
+                      __wt_compare(session, btree->collator, &cursor->key, &state.key, &exact));
```
Comment (luke-pearson) [inline src/btree/bt_cursor.c:1025]: I think this should be comparing against a bound not state.key which was a prefix search mechanism, I could be mistaken though.

Comment (luke-pearson) [reply src/btree/bt_cursor.c:1025]: Oh right because the search key set is state.key and we're only interested in knowing what side we're on.

Comment (luke-pearson) [reply src/btree/bt_cursor.c:1025]: Can you add a comment explaining why we need this extra compare in the bounded cursor case?

**Review by jiechenbo — APPROVED** [review]
LGTM!

File: src/btree/bt_cursor.c (hunk 1 above)
Comment (clarissecheah) [reply src/btree/bt_cursor.c:1031]: [5b00b50](https://github.com/wiredtiger/wiredtiger/pull/8231/commits/5b00b50496e7259fd3ca6808621d0aa24595aadf)

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Comment by clarissecheah** [comment]
evergreen merge

**Pull request merged**